gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -240,22 +215,47 @@ int main(int argc, char* argv[]) {
       return 3;
     }
 
-    auto proto = bhive_importer.BasicBlockProtoFromMachineCode(*bytes);
+    llvm::Expected<std::vector<gematria::DisassembledInstruction>>
+        instructions = gematria::DisassembleAllInstructions(
+            llvm_support->mc_disassembler(), llvm_support->mc_instr_info(),
+            llvm_support->mc_register_info(), llvm_support->mc_subtarget_info(),
+            *inst_printer, 0, *bytes);
 
     // Check for errors.
-    if (!proto.ok()) {
-      std::cerr << "Failed to disassemble block '" << hex << ": "
-                << proto.status() << "\n";
+    if (!instructions) {
+      std::cerr << "Failed to disassemble block '" << hex << "\n";
       continue;
     }
 
+    // Get used registers
+    std::map<unsigned, bool> used_registers;
+    for (const gematria::DisassembledInstruction& instruction : *instructions) {
+      for (unsigned operand_index = 0;
+           operand_index < instruction.mc_inst.getNumOperands();
+           ++operand_index) {
```

Comment:
With this, we enumerate inputs as well as outputs. Ideally, we should go from `instr_desc.getNumDefs()` to `mc_inst.getNumOperands()` to capture only the inputs.

Reply:
Good point. Switched over to iterating through the inputs of the register only (and keeping track of what registers get defined in the snippet so we can omit them from the usage list).